mixer lint errors still exist

Even with updating go runtime for linters to 1.9.x (see https://github.com/istio/istio/pull/1798), I see following (possibly new) lint errors for mixer. We should fix this.

```
mixer/adapter/svcctrl/distValueBuilder.go:42:1:warning: sizeOption is unused (deadcode)
mixer/adapter/svcctrl/handler.go:89:19:warning: method HandleApiKey should be HandleAPIKey (golint)
mixer/adapter/svcctrl/reportbuilder.go:148:1:warning: generateErrorCount is unused (deadcode)
mixer/adapter/svcctrl/reportbuilder.go:156:1:warning: generateBackendLatencies is unused (deadcode)
mixer/adapter/svcctrl/reportbuilder.go:37:1:warning: reportBuilder is unused (deadcode)
mixer/adapter/svcctrl/svcctrl.go:102:49:warning: error strings should not be capitalized or end with punctuation or a newline (golint)
mixer/adapter/svcctrl/svcctrl.go:108:16:warning: error strings should not be capitalized or end with punctuation or a newline (golint)
mixer/adapter/svcctrl/svcctrl.go:114:52:warning: error strings should not be capitalized or end with punctuation or a newline (golint)
mixer/adapter/svcctrl/svcctrl.go:125:25:warning: "postive" is a misspelling of "positive" (misspell)
mixer/adapter/svcctrl/svcctrl.go:43:19:warning: method SetApiKeyTypes should be SetAPIKeyTypes (golint)
mixer/adapter/svcctrl/svcctrl.go:80:49:warning: error strings should not be capitalized or end with punctuation or a newline (golint)
mixer/adapter/svcctrl/svcctrl.go:85:49:warning: error strings should not be capitalized or end with punctuation or a newline (golint)
mixer/adapter/svcctrl/testhelper.go:80:1:warning: mockConsumerProjectIDResolver is unused (deadcode)
mixer/adapter/svcctrl/testhelper.go:92:1:warning: compareJSON is unused (deadcode)
mixer/pkg/adapter/requestData_test.go:53:9:warning: should not use basic type int as key in context.WithValue (golint)
mixer/pkg/cache/cache.go:15:1:warning: package comment should be of the form "Package cache ..." (golint)
mixer/pkg/cache/cache_test.go:91::error: possible formatting directive in Error call (vet)
mixer/pkg/cache/lruCache.go:1:1:warning: package comment should be of the form "Package cache ..." (golint)
mixer/pkg/cache/ttlCache.go:1:1:warning: package comment should be of the form "Package cache ..." (golint)
mixer/pkg/config/handler_test.go:1::warning: file is not gofmted with -s (gofmt)
mixer/pkg/config/handler_test.go:1::warning: file is not goimported (goimports)
mixer/pkg/config/validator.go:1::warning: file is not gofmted with -s (gofmt)
mixer/pkg/config/validator.go:1::warning: file is not goimported (goimports)
mixer/pkg/config/validator.go:794::warning: line is 163 characters (lll)
mixer/pkg/expr/func.go:1::warning: file is not gofmted with -s (gofmt)
mixer/pkg/expr/func.go:1::warning: file is not goimported (goimports)
mixer/pkg/il/evaluator/checker.go:1::warning: file is not gofmted with -s (gofmt)
mixer/pkg/il/evaluator/checker.go:1::warning: file is not goimported (goimports)
mixer/pkg/il/evaluator/checker.go:38:1:warning: receiver name e should be consistent with previous receiver name c for checker (golint)
mixer/pkg/il/runtime/externs.go:1::warning: file is not gofmted with -s (gofmt)
mixer/pkg/il/runtime/externs.go:1::warning: file is not goimported (goimports)
mixer/pkg/il/runtime/externs.go:37:5:warning: exported var ExternFunctionMetadata should have comment or be unexported (golint)
mixer/pkg/il/runtime/externs.go:55:6:warning: func externIp should be externIP (golint)
mixer/pkg/il/runtime/externs.go:62:6:warning: func externIpEqual should be externIPEqual (golint)
mixer/pkg/il/testing/tests.go:32:5:warning: exported var TestData should have comment or be unexported (golint)
mixer/pkg/il/testing/util.go:26:10:warning: if block ends with a return statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary) (golint)
mixer/pkg/mock/server.go:30:6:warning: exported type Server should have comment or be unexported (golint)
mixer/tools/codegen/pkg/interfacegen/generator.go:35:2:warning: exported const ResourceMsgTypeSuffix should have comment (or a comment on this block) or be unexported (golint)
```

Some of the linter errors in svcctrl like this one
mixer/adapter/svcctrl/handler.go:89:19:warning: method HandleApiKey should be HandleAPIKey (golint)
is because the conflict between Go and proto message convention.  HandleApiKey is generated code, I don't think it is possible (or worth) to fix. Can we get an exemption on such thing?

--- a/lintconfig_base.json
+++ b/lintconfig_base.json
@@ -27,7 +27,6 @@
     ],
     "exclude": [
         "vendor",
-        "mixer/adapter/svcctrl",
         ".pb.go",
         "mixer/pkg/config/proto/combined.go",
         ".*.gen.go",

